crucible-llvm/src/Lang/Crucible/LLVM/Intrinsics/Common.hs

```diff
@@ -139,53 +139,58 @@ newtype ValTransformer p sym tp tp' =
 
 transformLLVMArgs :: forall m p sym bak args args'.
   (IsSymBackend sym bak, Monad m, HasLLVMAnn sym) =>
+  FunctionName ->
   bak ->
   CtxRepr args' ->
   CtxRepr args ->
   m (ArgTransformer p sym args args')
-transformLLVMArgs _ Ctx.Empty Ctx.Empty =
+transformLLVMArgs _fnName _ Ctx.Empty Ctx.Empty =
   return (ArgTransformer (\_ -> return Ctx.Empty))
-transformLLVMArgs bak (rest' Ctx.:> tp') (rest Ctx.:> tp) = do
+transformLLVMArgs fnName bak (rest' Ctx.:> tp') (rest Ctx.:> tp) = do
   return (ArgTransformer
            (\(xs Ctx.:> x) ->
-              do (ValTransformer f)  <- transformLLVMRet bak tp tp'
-                 (ArgTransformer fs) <- transformLLVMArgs bak rest' rest
+              do (ValTransformer f)  <- transformLLVMRet fnName bak tp tp'
+                 (ArgTransformer fs) <- transformLLVMArgs fnName bak rest' rest
                  xs' <- fs xs
                  x'  <- RegEntry tp' <$> f (regValue x)
                  pure (xs' Ctx.:> x')))
-transformLLVMArgs _ _ _ =
+transformLLVMArgs fnName _ _ _ =
   panic "Intrinsics.transformLLVMArgs"
-    [ "transformLLVMArgs: argument shape mismatch!" ]
+    [ "transformLLVMArgs: argument shape mismatch!" 
+    , "in function: " ++ Text.unpack (functionName fnName)
+    ]
 
 transformLLVMRet ::
   (IsSymBackend sym bak, Monad m, HasLLVMAnn sym) =>
+  FunctionName ->
```

Comment:
Similarly, state in the Haddocks that this `FunctionName` is only used for error-message purposes.